### PR TITLE
Move language resources out of `/guides` directory

### PR DESCRIPTION
**Pull request by parlough**
Contributes to https://github.com/dart-lang/site-www/issues/5767

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -36,17 +36,17 @@
       { "source": "/+isolates", "destination": "/language/concurrency", "type": 301 },
       { "source": "/+lexicalscope", "destination": "/language/functions#lexical-scope", "type": 301 },
       { "source": "/+pub", "destination": "/tools/pub", "type": 301 },
-      { "source": "/2.0", "destination": "/guides/language/evolution#dart-20", "type": 301 },
+      { "source": "/2.0", "destination": "/resources/language/evolution#dart-20", "type": 301 },
```
Comment (MaryaBelanger) [inline firebase.json:39]:
```suggestion
      { "source": "/2.0", "destination": "/resources/language/evolution#dart-2-0", "type": 301 },
```
Other one didn't work

File: firebase.json [hunk 2]
```diff
@@ -36,17 +36,17 @@
...
       { "source": "/\\)", "destination": "/", "type": 301 },
       { "source": "/angular{,/**}", "destination": "https://github.com/angulardart", "type": 301 },
       { "source": "/api", "destination": "https://api.dart.dev", "type": 301 },
       { "source": "/articles", "destination": "https://medium.com/dartlang", "type": 301 },
       { "source": "/articles/api-naming-guide{,/**}", "destination": "/effective-dart/design", "type": 301 },
       { "source": "/articles/archive", "destination": "https://medium.com/dartlang", "type": 301 },
       { "source": "/articles/archive/converters-and-codecs", "destination": "/libraries/convert/converters-and-codecs", "type": 301 },
-      { "source": "/articles/archive/dart-2", "destination": "/guides/language/evolution#dart-20", "type": 301 },
+      { "source": "/articles/archive/dart-2", "destination": "/resources/language/evolution#dart-20", "type": 301 },
```
Comment (MaryaBelanger) [inline firebase.json:47]:
```suggestion
      { "source": "/articles/archive/dart-2", "destination": "/resources/language/evolution#dart-2-0", "type": 301 },
```

File: firebase.json [hunk 3]
```diff
@@ -107,8 +107,8 @@
       { "source": "/code-of-conduct", "destination": "/community/code-of-conduct", "type": 301 },
       { "source": "/concurrency", "destination": "/language/concurrency", "type": 301 },
       { "source": "/dart", "destination": "/overview", "type": 301 },
-      { "source": "/dart-2", "destination": "/guides/language/evolution#dart-20", "type": 301 },
-      { "source": "/dart-2.0", "destination": "/guides/language/evolution#dart-20", "type": 301 },
+      { "source": "/dart-2", "destination": "/resources/language/evolution#dart-20", "type": 301 },
+      { "source": "/dart-2.0", "destination": "/resources/language/evolution#dart-20", "type": 301 },
```
Comment (MaryaBelanger) [inline firebase.json:111]:
```suggestion
      { "source": "/dart-2", "destination": "/resources/language/evolution#dart-2-0", "type": 301 },
      { "source": "/dart-2.0", "destination": "/resources/language/evolution#dart-2-0", "type": 301 },
```

**Review by MaryaBelanger — APPROVED** [review]

**Comment by parlough** [comment]
Thanks for catching those broken fragments @MaryaBelanger!

**Pull request merged**